lib/rules/ruleInstantiations.ts

```diff
@@ -8,24 +8,88 @@ export class RuleInstantiation extends RuleBase implements IRule {
+
+        }
+      }
+    }
+    // Filter all interface Elements
+    const fittingInterfaceElements = [];
+    const incompleteInterfaceElements = [];
+    for (const interfaceElement of availableInterfaceElements) {
+      let valid = Object.keys(namedArguments).every(argument => interfaceElement.some(port => port.lexerToken?.getLText() === argument));
+      if (interfaceElement.length - Object.keys(namedArguments).length < numberedArguments.length) {
+        valid = false;
+      }
+      if (valid === false) {
+        continue;
+      }
+      const missingPorts = [];
+      for (const port of interfaceElement.slice(numberedArguments.length)) {
+        if (port instanceof O.OTypeMark || port.defaultValue) {
           continue;
         }
-        if (association.formalPart.length > 1) {
+        if (Object.keys(namedArguments).includes(port.lexerToken.getLText()) === false) {
+          missingPorts.push(port);
+        }
+      }
+      if (missingPorts.length > 0) {
+        incompleteInterfaceElements.push(missingPorts);
+      } else {
+        fittingInterfaceElements.push(interfaceElement);
+      }
+    }
+    if (fittingInterfaceElements.length > 0) { // No error some are fitting
+      // Check for open without default
+      for (const association of associationList?.children ?? []) {
+        if (association.actualIfInput.length === 1 && association.actualIfInput[0]!.nameToken.getLText() === 'open'
+          && association.definitions.every(def => (def instanceof O.OPort && def.direction === 'in' && def.defaultValue === undefined || def instanceof O.OGenericConstant && def.defaultValue === undefined))) {
           this.addMessage({
             range: association.range,
-            severity: DiagnosticSeverity.Warning,
-            message: `Multiple formal references parsed (${association.formalPart.map(ref => ref.nameToken.text).join(', ')}). Problem likely`
+            severity: DiagnosticSeverity.Error,
+            message: association.parent instanceof O.OPortAssociationList ?
+              `port ${association.formalPart[0]?.nameToken.text ?? 'unknown'} is an input without default value and must not be open.` :
+              `generic ${association.formalPart[0]?.nameToken.text ?? 'unknown'} has no default value and must not be open.`
           });
         }
-        allElementsWithoutFormal = false;
+      }
+    } else if (incompleteInterfaceElements.length > 0) {
+      const shortestMissing = incompleteInterfaceElements.reduce((ports, missingPortsThis) => {
+        if (ports.length < missingPortsThis.length) {
+          return ports;
+        }
+        return missingPortsThis;
+      }, incompleteInterfaceElements[0]!);
+
+      this.addMessage({
+        range: range,
+        severity: DiagnosticSeverity.Error,
+        message: `Missing connection for ports ${shortestMissing.map(port => `'${port.lexerToken.text}'`).join(', ')}`
```

Comment:
can also be generics?
```suggestion
        message: `Missing connection for _ports_ ${shortestMissing.map(port => `'${port.lexerToken.text}'`).join(', ')}`
```